scanpy/tests/test_plotting.py

```diff
@@ -1215,6 +1215,76 @@ def test_scatter_specify_layer_and_raw():
         sc.pl.umap(pbmc, color="HES4", use_raw=True, layer="layer")
 
 
+def test_scatter_no_basis_per_obs(image_comparer):
+    """Test scatterplot of per-obs points with no basis"""
+    save_and_compare_images = image_comparer(ROOT, FIGS, tol=15)
+    pbmc = sc.datasets.pbmc68k_reduced()
+    sc.pl.scatter(pbmc, x="HES4", y="percent_mito", color="n_genes", use_raw=False)
+    save_and_compare_images("scatter_HES_percent_mito_n_genes")
+
+
+def test_scatter_no_basis_per_var(image_comparer):
+    """Test scatterplot of per-var points with no basis"""
+    save_and_compare_images = image_comparer(ROOT, FIGS, tol=15)
+    pbmc = sc.datasets.pbmc68k_reduced()
+    sc.pl.scatter(pbmc, x="AAAGCCTGGCTAAC-1", y="AAATTCGATGCACA-1", use_raw=False)
+    save_and_compare_images("scatter_AAAGCCTGGCTAAC-1_vs_AAATTCGATGCACA-1")
+
+
+@pytest.fixture
+def pbmc_filtered():
+    pbmc = sc.datasets.pbmc68k_reduced()
+    sc.pp.filter_genes(pbmc, min_cells=10)
+    return pbmc
+
+
+def test_scatter_no_basis_raw(check_same_image, pbmc_filtered):
+    """Test scatterplots of raw layer with no basis."""
+    path1 = FIGS / "scatter_EGFL7_F12_FAM185A_rawNone.png"
+    path2 = FIGS / "scatter_EGFL7_F12_FAM185A_rawTrue.png"
+    path3 = FIGS / "scatter_EGFL7_F12_FAM185A_rawToAdata.png"
+
+    sc.pl.scatter(pbmc_filtered, x='EGFL7', y='F12', color='FAM185A', use_raw=None)
+    plt.savefig(path1)
+    plt.close()
+
+    # is equivalent to:
+    sc.pl.scatter(pbmc_filtered, x='EGFL7', y='F12', color='FAM185A', use_raw=True)
+    plt.savefig(path2)
+    plt.close()
+
+    check_same_image(path1, path2, tol=15)
+
+    # and also to:
+    sc.pl.scatter(pbmc_filtered.raw.to_adata(), x='EGFL7', y='F12', color='FAM185A')
+    plt.savefig(path3)
```

Comment:
```suggestion
    # and also to:
    sc.pl.scatter(pbmc_filtered.raw.to_adata(), x='EGFL7', y='F12', color='FAM185A')
    plt.savefig(path3)
    
    check_same_image(path1, path2, tol=15)
```

This way we'll have up to date output for path3 if the first assertion fails – would be useful info for debugging.